packages/aws_common/test/logging/aws_logger_test.dart

```diff
@@ -220,5 +222,72 @@ void main() {
+      final plugin = MockLoggerPlugin();
+      final logger = AWSLogger()..registerPlugin(plugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), plugin);
+
+      final childLogger = logger.createChild('child');
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), plugin);
+
+      logger.unregisterPlugin(plugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), isNull);
+
+      childLogger.registerPlugin(plugin);
+      expect(logger.getPlugin<MockLoggerPlugin>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), plugin);
+    });
+
+    test(
+        'registerPlugin handles registering one plugin per type in the logger hierarchy',
+        () {
+      final plugin = MockLoggerPlugin();
+      final childLogger = AWSLogger()
+          .createChild('child')
+          .createChild('grandChild')
+          .createChild('grandGrandChild');
+      expect(() => childLogger.registerPlugin(plugin), returnsNormally);
+      expect(() => childLogger.registerPlugin(plugin), throwsStateError);
+      expect(() => AWSLogger().registerPlugin(plugin), throwsStateError);
+
+      final newPlugin = MockLoggerPlugin();
+      expect(() => AWSLogger().registerPlugin(newPlugin), throwsStateError);
+      expect(() => childLogger.registerPlugin(newPlugin), throwsStateError);
+
+      expect(() => childLogger.unregisterPlugin(plugin), returnsNormally);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), isNull);
+
+      expect(() => AWSLogger().registerPlugin(newPlugin), returnsNormally);
+      expect(() => childLogger.registerPlugin(newPlugin), throwsStateError);
+      expect(AWSLogger().getPlugin<MockLoggerPlugin>(), newPlugin);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), newPlugin);
+    });
+
+    test('registerPlugin allows registering plugins of same superclass', () {
+      final logger = AWSLogger();
+      final plugin1 = MockLoggerPlugin();
+      final plugin2 = MockLoggerPlugin2();
+
+      // test registering plugins to the logger itself.
+      expect(() => logger.registerPlugin(plugin2), returnsNormally);
+      expect(() => logger.registerPlugin(plugin1), returnsNormally);
+
+      expect(logger.getPlugin<MockLoggerPlugin>(), plugin1);
+      expect(logger.getPlugin<MockLoggerPlugin2>(), plugin2);
+
+      logger.unregisterAllPlugins();
+
+      // test registering plugins to the logger and its child.
+      final childLogger = logger.createChild('child');
+      expect(() => childLogger.registerPlugin(plugin2), returnsNormally);
+      expect(() => logger.registerPlugin(plugin1), returnsNormally);
+
+      expect(logger.getPlugin<MockLoggerPlugin>(), plugin1);
+      expect(logger.getPlugin<MockLoggerPlugin2>(), isNull);
+      expect(childLogger.getPlugin<MockLoggerPlugin>(), plugin1);
+      expect(childLogger.getPlugin<MockLoggerPlugin2>(), plugin2);
+    });
```

Comment:
Trailing comma will help with formatting. Generally, if test names go over 80chars it's best to wrap them using adjacent strings.